### PR TITLE
Feat: Automatic releases after build job

### PR DESCRIPTION
**Pull request by cawolfkreo**

### CODE REVIEW

**Comment by cawolfkreo** [comment]
@seansaleh I squashed all of the commits into one and now releases work. You can see examples of manual and automatic builds on [the releases section of the repo](https://github.com/seansaleh/artifact-manager-s3-plugin/releases). Hope this is working as you expected!!

File: .github/workflows/maven-automatic.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    - name: Create release
+      uses: "marvinpinto/action-automatic-releases@latest"
+      with:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+        automatic_release_tag: 'latest'
+        prerelease: ${{ env.GITHUB_EVENT_NAME == 'pull_request' }}
+        draft: false
+        title: "Latest Release Build"
+        files: |
+          *.hpi
```
Comment (seansaleh) [inline .github/workflows/maven-automatic.yml:61]: We actually wouldn't want to auto create a release on every commit. Instead we'd just want to run the first steps (build & package)

File: .github/workflows/maven-publish.yml [hunk 2]
```diff
@@ -11,14 +11,26 @@ on:
         required: false
         default: "1145.vb_cf6cf6ed960"
         type: string
-  push:
-    branches: [master]
-  pull_request:
+      createRelease:
+        description: "Wether or not the build should trigger a new release on Github"
```
Comment (seansaleh) [inline .github/workflows/maven-publish.yml:15]: Based on how this shows up we should just call it "Create Github Release" 
image

File: .github/workflows/maven-automatic.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#apache-maven-with-a-settings-path
+
+name: Maven Automatic Package
```
Comment (seansaleh) [inline .github/workflows/maven-automatic.yml:4]: We should come up with more descriptive names.

Maybe "Automatic Build & Package"
And the other job is "Manual Build, Package & Release"

**Review by seansaleh — COMMENTED** [review]
A few small changes requested.
Also, if you can go back and squash any CI/CD related commits into one, then when merging the branch make sure to use a squash commit, not a merge commit

**Comment by seansaleh** [comment]
Overall, also the version number should be in the release info somewhere, potentially as a git/release tag

**Comment by seansaleh** [comment]
It also looks like the version doesn't properly get set. Inspecting the .hpi file at `artifact-manager-s3.hpi\META-INF\maven\io.jenkins.plugins\artifact-manager-s3\pom.xml` has `<version>999999-SNAPSHOT</version>`, and this is the version that shows up in Jenkins :(
image

**Pull request merged**